stl/src/multprec.cpp

```diff
@@ -90,16 +90,16 @@ static void div(_MP_arr u,
     unsigned long long
         v0) noexcept { // divide multi-word value by scalar value (fits in lower half of unsigned long long)
     unsigned long long k = 0;
-    int ulen             = _MP_len;
-    while (0 <= --ulen) { // propagate remainder and divide
-        unsigned long long tmp = (k << shift) + u[ulen];
-        u[ulen]                = tmp / v0;
-        k                      = tmp % v0;
+    for (int ulen = _MP_len - 1; ulen >= 0; --ulen) { // propagate remainder and divide
+        unsigned long long t = (k << shift) | u[ulen];
+        u[ulen]              = t / v0;
+        k                    = t - (u[ulen] * v0); // a mod b = a - (floor(a / b) * b)
     }
 }
 
 _NODISCARD static int limit(const unsigned long long* u, int ulen) noexcept { // get index of last non-zero value
-    while (u[ulen - 1] == 0) {
+    // Prevent from reading outside of ulen if 0 is not found
+    while (ulen > 0 && u[ulen - 1] == 0) {

```

Comment:
This code is a bit slower because of the extra check. However, I cannot prove that every time this function is called, there won't be an out of bounds read, so I did this to be safe.